docs/proposals/simulation/simulation.md

```diff
@@ -0,0 +1,165 @@
+
+1. Add `self.dataset_url`, `self.dataset_path`
+2. Add `download_save()` function
+
+![](https://github.com/iszhyang/images/blob/main/ianvs/dataset-download.png?raw=true)
+
+### Semantic segmentation dataset and corresponding algorithm
+
+dataset: [cityscapes](https://www.cityscapes-dataset.com)
+Baseline algorithm: [RFNet](https://github.com/AHupuJR/RFNet)
+
+We will provide the cityscapes dataset and corresponding methods for ianvs. The specific operation is as follows:
+
+- dataset processing
+  - generate the index file for train and test
+- dataset download link
+  - provide the download link of cityscapes dataset in HUAWEI Cloud OBS.
+- Baseline algorithm
+  ![](https://github.com/iszhyang/images/blob/main/ianvs/rfnet.png?raw=true)
+
+### Industrial Distributed Collaborative System Simulation
+
+Provide ianvs with the feature of industrial distributed system simulation.
+#### Framework
+![img](images/simulation_framework.jpg)
+
+In the framwork of ianvs, the `simulation controller` is the core module of system simulation. The `simulation controller` has been supplemented, which build and deploy local edge-cloud simulation environment with K8s.
+
+![img](images/simulation_controller.jpg)
+
+The models in `simulation controller` are as follows:
+- The `Simulation System Administrator` is used to
+  1. parse the system config(simulation)
+  2. build the simulation enviroment
+  3. create and deploy the moudles needed in simulation enviroment
+  3. close and delete the simulation enviroment
+- The `Simulation Job Administrator` is the core module for manage the simulation job, and provides the following funcitons:
+  1. generate the YAML file of `simulation job`
+  2. deploy and delete the `simulation job` in K8s
+  3. list-watch the results of `simulation job` in K8s
+
+#### Dataflow
+![img](images/simulation_dataflow.jpg)
+
+The `System Config` denotes the system config of the current benchmarkingjob, such as the `cloud number` and `edge number`.
+
+In the data flow diagram above, the expected flow is as follows:
+1. start the benchmarkingjob
+2. check whether to start emulation based on `system config`
+3. `Simulation System Administrator` check environment and parse system config
+  - check environment: eg. check if the host has installed docker, kind, and whether the memory > 4GB
+  - parse simulation config
+4. `Simulation System Administrator` build the environment and create needed moudles.
+  - build the environment: deploying a cluster including K8s and GM (global manager) locally through [all-in-one scripts of sedna](https://github.com/kubeedge/sedna/blob/527c574a60d0ae87b0436f9a8b38cf84fb6dab21/docs/setup/all-in-one.md)
+  - create and deploy the `simulation job controller`
+5. `Simulation Job Administrator` generate and deploy simulation jobs.
+  - generate the YAML of simulation job base on `testenv.yaml` and `algorithm.yaml`
```

Comment:
Add a step here: we need to help algorithm developers build algorithms into docker images.

Reply:
Thanks. We do need to help users package the algorithms under test into docker image. I have made modifications to the process and flowchart.